doc: add AWS SDK checksum configuration tips for S3 compatibility (#7177)
juicedata/juicefs: docs/en/reference/how_to_set_up_object_storage.md

The format of the option `--bucket` for all S3 compatible object storage services is `https://<bucket>.<endpoint>` or `https://<endpoint>/<bucket>`. The default `region` is `us-east-1`. When a different `region` is required, it can be set manually via the environment variable `AWS_REGION` or `AWS_DEFAULT_REGION`.
:::

:::tip

> For AWS SDK request and response checksums, JuiceFS defaults `AWS_REQUEST_CHECKSUM_CALCULATION` and `AWS_RESPONSE_CHECKSUM_VALIDATION` to `when_required`. This avoids sending or validating SDK-level checksum headers unless S3 requires them, which improves compatibility with S3-compatible services.

Reviewer: ```suggestion
For AWS SDK request and response checksums, JuiceFS sets `AWS_REQUEST_CHECKSUM_CALCULATION` and `AWS_RESPONSE_CHECKSUM_VALIDATION` to `when_required` by default. As a result, SDK-level checksum headers are only sent or validated when required by S3. This improves compatibility with S3-compatible services.
```

> If you use Amazon S3 and want the SDK to calculate and validate checksums whenever the operation supports them, set the environment variables before running JuiceFS:

Reviewer: ```suggestion
If you are using Amazon S3 and want the SDK to calculate and validate checksums whenever checksum support is available for an operation, set the following environment variables before running JuiceFS:
```

```shell
export AWS_REQUEST_CHECKSUM_CALCULATION=when_supported
export AWS_RESPONSE_CHECKSUM_VALIDATION=when_supported
```

> This only controls AWS SDK-level checksums. The `disable-checksum=true` bucket query option controls JuiceFS' own CRC checksum stored in object metadata.

Reviewer: ```suggestion
These settings only affect AWS SDK-level checksums. The `disable-checksum=true` query parameter in `--bucket` controls the CRC checksum that JuiceFS stores in object metadata.
```
